### PR TITLE
Make allora-producer updates less error-prone

### PR DESCRIPTION
**Pull request by fernandofcampos**
- Created a CONTRIBUTING.md file to document how to update the producer when allora-chain changes
- Created tests to ensure that all versions of the types of transactions and events are supported (this test would have avoided the issue we had with the last upgrade)
- Migrated to use the registration functions from the chain, and removed the local files that are not needed anymore (we won't need to copy files from the chain from now on either).
- Added some missing event files (need to check if we can include events on registration functions at the chain, so we don't need to keep those anymore)
- Few small improvements to make upgrades easier

The goal is to prevent the errors that happened during the last upgrade (solved here: https://github.com/allora-network/allora-producer/pull/21)

### CODE REVIEW

File: app/service/query.go [hunk 1]
```diff
@@ -13,6 +13,11 @@ import (
 	"github.com/cosmos/gogoproto/proto"
 )
 
+const (
+	EmissionsServicePrefix = "/emissions.v7.QueryService/"
+	MintServicePrefix      = "/mint.v5.QueryService/"
```
Comment (felipead) [inline app/service/query.go:18]: Well done!

File: docker-compose.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ services:
       - ALLORA_KAFKA_USER=${ALLORA_KAFKA_USER}
       - ALLORA_KAFKA_PASSWORD=${ALLORA_KAFKA_PASSWORD}
       - ALLORA_QUERY_RESULT_REPOSITORY_URL=${ALLORA_QUERY_RESULT_REPOSITORY_URL}
-      - ALLORA_API_KEY=${ALLORA_API_KEY}
+      - ALLORA_ALLORA_API_API_KEY=${ALLORA_ALLORA_API_API_KEY}
```
Comment (felipead) [inline docker-compose.yml:14]: Why this? 🤔

File: codec/codec_test.go [hunk 3]
```diff
@@ -42,13 +44,38 @@ func TestCodec_AddInterfaces(t *testing.T) {
...
+
+		// MINT TX TYPES
+		"/mint.v1beta1.MsgUpdateParams",
+		"/mint.v2.UpdateParamsRequest",
+		"/mint.v5.UpdateParamsRequest",
+	}
+
+	for _, txType := range txTypes {
+		_, err := c.interfaceRegistry.Resolve(txType)
+		require.NoError(t, err)
```
Comment (felipead) [inline codec/codec_test.go:75]: 👏

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    3. Update `TestCodec_DefaultRegisterFuncs` test on `codec/codec_test.go`, inserting the new types (e.g. vB) to ensure full coverage.
+    4. Update the config files under `config/` with the new versions (e.g. vB)
+    5. Update `app/service/query.go`
+        1. Update `EmissionsServicePrefix` and `MintServicePrefix` constants
+        2. Check if the names of the types did not change
+3. If a new version of Event types were created (e.g. vA gets updated to vB):
+    1. Copy the old version of `events.pb.go` under `codec/allora-chain/x` (follow the structure of existing versions)
+    2. Add a silent import on codec/codec.go` (at "OLD VERSIONS of MINT EVENTS" or "OLD VERSIONS of EMISSIONS EVENTS" sections)
+    3. Add a new test at `TestAllEventsAreRegistered`, on `tests/integration/codec/codec_test.go` to ensure all event types are recognized.
+    4. Update the config files under `config/` with the new versions (e.g. vB)
```
Comment (felipead) [inline CONTRIBUTING.md:37]: thank you for this

**Review by felipead — APPROVED** [review]

File: docker-compose.yml (hunk 2 above)
Comment (fernandofcampos) [reply docker-compose.yml:14]: This was wrong. This file is only used locally, that's why we did not catch it.

File: CONTRIBUTING.md (hunk 4 above)
Comment (KjetilVaa) [reply CONTRIBUTING.md:37]: Nice!

**Review by KjetilVaa — APPROVED** [review]
LGTM

**Pull request merged**